DataPlotly/core/plot_types/plot_type.py

```diff
@@ -59,12 +59,35 @@ def create_layout(settings):
             x_title = settings.layout['x_title']
             y_title = settings.layout['y_title']
 
+        range_x = None
+        if 'x_min' in settings.layout and 'x_max' in settings.layout:
+            if settings.layout['x_min'] is not None and settings.layout['x_max'] is not None:
```

Comment:
This could be simplified:

    ```suggestion
            if settings.layout.get('x_min', None) is not None and settings.layout.get('x_max', None) is not None:
```

Reply:
Done!